janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java

```diff
@@ -252,21 +252,25 @@ Cluster initializeCluster() throws PermanentBackendException {
     }
 
     Session initializeSession(final String keyspaceName) {
-        final Configuration configuration = getStorageConfig();
-        final Map<String, Object> replication = Match(configuration.get(REPLICATION_STRATEGY)).of(
+        final Session s = this.cluster.connect();
+
+        if (this.cluster.getMetadata().getKeyspace(keyspaceName) == null) {
```

Comment:
do the check for not null first and return so that you can un-nest the large conditional block?

Reply:
yep, will do.